system-test/bigtable.ts

```diff
@@ -1414,6 +1421,269 @@ describe('Bigtable', () => {
 
       Object.keys(policy).forEach(key => assert(key in updatedPolicy));
     });
+    describe('copying backups', () => {
+      const sourceExpireTimeMilliseconds =
+        PreciseDate.now() + (8 + 300) * 60 * 60 * 1000;
+      const sourceExpireTime = new PreciseDate(sourceExpireTimeMilliseconds);
+      const copyExpireTimeMilliseconds =
+        PreciseDate.now() + (8 + 600) * 60 * 60 * 1000;
+      const copyExpireTime = new PreciseDate(copyExpireTimeMilliseconds);
+
+      /*
+        This function checks that when a backup is copied using the provided
+        config that a new backup is created on the instance.
+       */
+      async function testCopyBackup(
+        backup: Backup,
+        config: CopyBackupConfig,
+        instance: Instance
+      ) {
+        // Get a list of backup ids before the copy
+        const [backupsBeforeCopy] = await instance.getBackups();
+        const backupIdsBeforeCopy = backupsBeforeCopy.map(backup => backup.id);
+        // Copy the backup
+        const [newBackup, operation] = await backup.copy(config);
+        assert.strictEqual(config.id, newBackup.id);
+        await operation.promise();
+        const id = config.id;
+        const backupPath = `${config.cluster.name}/backups/${id}`;
+        {
+          // Ensure that the backup specified by the config and id match the backup name for the operation returned by the server.
+          // the split/map/join functions replace the project name with the {{projectId}} string
+          assert(operation);
+          assert(operation.metadata);
+          assert.strictEqual(
+            operation.metadata.name
+              .split('/')
+              .map((item, index) => (index === 1 ? '{{projectId}}' : item))
+              .join('/'),
+            backupPath
+              .split('/')
+              .map((item, index) => (index === 1 ? '{{projectId}}' : item))
+              .join('/')
+          );
+        }
+        // Check that there is now one more backup
+        const [backupsAfterCopy] = await instance.getBackups();
+        const newBackups = backupsAfterCopy.filter(
+          backup => !backupIdsBeforeCopy.includes(backup.id)
+        );
+        assert.strictEqual(newBackups.length, 1);
+        const [fetchedNewBackup] = newBackups;
+        // Ensure the fetched backup matches the config
+        assert.strictEqual(fetchedNewBackup.id, id);
+        assert.strictEqual(fetchedNewBackup.name, backupPath);
+        // Delete the copied backup
+        await config.cluster.backup(fetchedNewBackup.id).delete();
```

Comment:
If this test fails, do you need to worry about backups being leftover? Or is the entire resource destroyed no matter what?

Reply:
I added better cleanup for the backups both created from scratch and copied.